server/block/grindstone.go

```diff
@@ -0,0 +1,100 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Grindstone is a block that repairs items and tools as well as removing enchantments from them. It also serves as a
+// weaponsmith's job site block.
+type Grindstone struct {
+	transparent
+
+	// Attach represents the attachment type of the Grindstone.
+	Attach GrindstoneAttachment
+	// Direction represents the direction the Grindstone is facing.
+	Direction cube.Direction
+}
+
+// BreakInfo ...
+func (g Grindstone) BreakInfo() BreakInfo {
+	return newBreakInfo(2, pickaxeHarvestable, pickaxeEffective, oneOf(g)).withBlastResistance(30)
+}
+
+// Activate ...
+func (g Grindstone) Activate(pos cube.Pos, _ cube.Face, _ *world.World, u item.User, _ *item.UseContext) bool {
+	if opener, ok := u.(ContainerOpener); ok {
+		opener.OpenBlockContainer(pos)
+		return true
+	}
+	return false
+}
+
+// UseOnBlock ...
+func (g Grindstone) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.World, user item.User, ctx *item.UseContext) (used bool) {
+	pos, face, used = firstReplaceable(w, pos, face, g)
+	if !used {
+		return false
+	}
+	if face == cube.FaceUp {
+		g.Direction = user.Facing().Opposite()
```

Comment:
Can move `g.Direction = user.Facing().Opposite()` in front of the `if`s and only change it in the `else` case to save a line.